client/comms/wsconn_test.go

```diff
@@ -87,17 +104,22 @@ func TestWsConn(t *testing.T) {
 
 		c, err := upgrader.Upgrade(w, r, nil)
 		if err != nil {
-			t.Fatalf("unable to upgrade http connection: %s", err)
+			t.Errorf("unable to upgrade http connection: %s", err)
 		}
 
+		ch := &conn{Conn: c}
+		clientMtx.Lock()
+		clients[id] = ch
+		clientMtx.Unlock()
+
 		c.SetPongHandler(func(string) error {
 			t.Logf("handler #%d: pong received", id)
```

Comment:
This t.Logf inside c.SetPongHandler was where the race occasionally occurred because connection wasn't stopped and the next test started with it (and SetPongHandler) still running.